src/data/stackexchange/stack_exchange_parse.py

```diff
@@ -278,30 +224,36 @@ def main(args):
 
     if args.skipparse == False:
         for (section, url) in stack_exchange_data:
-            print("Starting " + section)
-
             #creates directories for the current SE site
             zip_file_path, unzipped_folder, corpus_section_directory = section_setup(
                 section, zip_directory, corpus_directory)
 
+            if os.path.isfile(os.path.join(corpus_section_directory, ".done")):
+                continue
+
+            print("Starting " + section)
+
+
             #downloads and unzips data release for a site
             load(url, zip_file_path, unzipped_folder)
 
             #gets the links data from the links table for the site
             links = get_links(unzipped_folder)
 
-            #creates the clusters of related and duplicate posts for a site,
-            #based on links data
-            clusters, related, duplicates, unique_posts = gen_clusters(links)
-
             #gets post data from the posts table
             posts = get_posts(unzipped_folder)
 
-            #extract post title and body text for each document in the site
-            corpus = gen_corpus(posts, unique_posts)
+            #creates the clusters of related and duplicate posts for a site,
+            #based on links data
+            # clusters, related, duplicates, unique_posts = gen_clusters(links)
+            clusters = iter_clusters(links, posts)
 
             #writes cluster information to json files
-            write_json_files(clusters, related, duplicates, corpus, corpus_section_directory)
+            write_json_files(clusters, corpus_section_directory)
+            
+            # put completion marker in folder so we can skip it next time
+            with open(os.path.join(corpus_section_directory, ".done"), "w") as f:
+                print("", file=f)
 
             print("Completed " + section)
 
```

Comment:
The next step was built to filter the json files to only retain clusters that had between 3 and 10 posts. Given the new paired posting logic, I believe that the filter command line argument and the filter_json_files function are no longer needed.